lib/spack/spack/binary_distribution.py

```diff
@@ -748,6 +748,7 @@ def generate_package_index(cache_prefix):
             # s = Spec.from_yaml(yaml_obj)
             s = Spec.from_yaml(yaml_contents)
             db.add(s, None)
+            db.mark(s, 'cache_exists', True)
```

Comment:
do we want a separate `mark()` routine or just an optional dictionary arg passed with `add()`?  Asking b/c they're only used together right now, and this is two write transactions (if not nested in a larger one, which I believe this is).

I think it will be easy to forget that lock/read/write/unlock twice, accidentally, but maybe it's useful to have the method separate.  Thoughts?

Reply:
I think it's useful to have as a separate method because of the way `add` is recursive. I find the arguments that we set for the top level of recursion and then pop off to some default for the lower levels difficult to think about, so I wanted to avoid implementing it that way (which was my first thought as well).

Reply:
ok let's stick with it.  If the transactions are nested I think we are good.